### PR TITLE
Quick Examples for the database package

### PR DESCRIPTION
**Pull request by Punyaslok**
An introduction to what the database package offers.

Also contains commits from #1785 and #1871

### CODE REVIEW

File: examples/gallery/database_module.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+##############################################################################
+# First we need to instantiate a `~sunpy.database.Database` object. Note that
+# the argument can be any URL supported by SQLAlchemy.
+
+database = Database('sqlite:///sunpydata.sqlite')
+
+##############################################################################
+# Adding Entries from a VSO query
+
+vso_query_result = vso.VSOClient().query(
```
Comment (Cadair) [inline examples/gallery/database_module.py]: I would instanciate the client on a seperate line.

File: examples/gallery/database_module.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
+	vso.attrs.Instrument('AIA'))
+database.add_from_vso_query_result(vso_query_result)
```
Comment (Cadair) [inline examples/gallery/database_module.py]: you need to make it clear that this does not download the data.

File: examples/gallery/database_module.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Downloading data. We can download the required files and store them so
+# that they can be easily accessed for future use. An optional argument path
+# can be passed to specify the download location.
+
+database.download(vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
```
Comment (Cadair) [inline examples/gallery/database_module.py]: Will this replace the records added above? and add file paths? either way you need to add more description about what is happening inside the database.

File: examples/gallery/database_module.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+##############################################################################
+# Emptying the database. The clear() method can be used to remove all the
+# entries in the database.
+
+database.clear()
+print(len(database))
+
+##############################################################################
+# Committing changes to the database. While working in an interactive python
```
Comment (Cadair) [inline examples/gallery/database_module.py]: you have to commit always right?

File: examples/gallery/database_module.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+vso_query_result = vso.VSOClient().query(
+	vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
+	vso.attrs.Instrument('AIA'))
+database.add_from_vso_query_result(vso_query_result)
+
+##############################################################################
+# Now to see what is in the database, we use the display_entries function to
+# display the desired parameters in a neat way.
+
+print(display_entries(database,
```
Comment (Cadair) [inline examples/gallery/database_module.py]: I think `display_entries` should have a sane set of deaults, it seems stupid to specify this list everytime when 90% of the time you want the same columns.

File: examples/gallery/database_module.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+# has already been used once to add entries using the download() method.
+# Otherwise, the given query is used to download and add new data via the
+# download() method. This means that if you are not sure whether the required
+# data already exists in the database, use fetch() to save bandwidth.
+
+print("Number of entries in database : " + str(len(database)))
+
+##############################################################################
+
+entries = database.fetch(vso.attrs.Time('2011-05-08', '2011-05-08 00:00:05'),
```
Comment (Cadair) [inline examples/gallery/database_module.py]: on a higher level, I am not sure I like the distinction between the `download` and `fetch` methods. It get's confusing to me, so I can't imaging that it's clear to end users.

File: examples/gallery/database_module.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+# added to the database. If you are not sure whether the required data already
+# exists in the database, use the fetch() method !!!
+
+##############################################################################
+# Changing an entry's atrributes. We can use the Database.edit() method to
+# change any attribute of a database entry. Here we edit the
+# wavemax attribute ( if it is less than 20.0, then we change it to 100.0;
+
+for database_entry in database:
+	if database_entry.wavemax < 20.0:
```
Comment (Cadair) [inline examples/gallery/database_module.py]: I feel like this should be using astropy units.

File: examples/gallery/database_module.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+##############################################################################
+# Starring Entries. We can star(italic) database entries which are of interest
+# to us. Here we star all those entries with wavemin greater than 20.
+
+for database_entry in database:
+	if database_entry.wavemin > 20:
+		database.star(database_entry)
+
+print(display_entries(
+	filter(lambda entry: entry.starred, database),
```
Comment (Cadair) [inline examples/gallery/database_module.py]: this needs some more description. I am not even sure what this is doing!

File: examples/gallery/database_module.py (hunk 4 above)
Comment (Punyaslok) [inline examples/gallery/database_module.py]: Commit is only required when you want to save the changes made to the database. Only after committing, the database file on the disk will be updated.

File: examples/gallery/caching_in_database.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+# Import the necessary modules.
+
+from sunpy.database import Database
+from sunpy.database.tables import display_entries
+from sunpy.net import vso
+
+##############################################################################
+# Instantiate the VSO and database clients
+
+client = vso.VSOClient()
```
Comment (Cadair) [inline examples/gallery/caching_in_database.py]: Is this client line needed?

File: examples/gallery/displaying_database.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# The contents can also be printed simply by using print statement on the
+# database object. Note that the resultant table may be truncated according to
+# the terminal size.
+
+print (database)
```
Comment (Cadair) [inline examples/gallery/displaying_database.py]: No space

**Comment by Punyaslok** [comment]
This needs modification now that [#1785](https://github.com/sunpy/sunpy/pull/1785) has changed quite a bit.

**Pull request closed without merging**